providers/google.go

```diff
@@ -98,17 +101,21 @@ func NewGoogleProvider(p *ProviderData, opts options.GoogleOptions) (*GoogleProv
 		},
 	}
 
+	var file *os.File
 	if opts.ServiceAccountJSON != "" {
-		file, err := os.Open(opts.ServiceAccountJSON)
+		var err error
+		file, err = os.Open(opts.ServiceAccountJSON)
 		if err != nil {
 			return nil, fmt.Errorf("invalid Google credentials file: %s", opts.ServiceAccountJSON)
 		}
+	}
 
+	if opts.ServiceAccountJSON != "" || opts.UseApplicationDefaultCredentials {
 		// Backwards compatibility with `--google-group` option
 		if len(opts.Groups) > 0 {
 			provider.setAllowedGroups(opts.Groups)
 		}
```

Comment:
as one or the other has to be set for the provider in the validation phase already this check seems unnecessary and the content of this block will always be executed anyway. therefore i suggest to removing the condition

```suggestion
	// Backwards compatibility with `--google-group` option
	if len(opts.Groups) > 0 {
		provider.setAllowedGroups(opts.Groups)
	}
	
	provider.setGroupRestriction(opts)
```